gravitee-io/gravitee-api-management
# Remove calls to `GraviteeContext` from Services - Part 1

**Issue**

Related to:
 - https://github.com/gravitee-io/issues/issues/6071
 - https://github.com/gravitee-io/gravitee-api-management/pull/854

Fix: 
 - https://github.com/gravitee-io/issues/issues/6488
 - Create an OrganizationAuditLog instead of an Env one when dealing with Entrypoint https://github.com/gravitee-io/gravitee-api-management/pull/971/commits/7dd4d78ed29fdaafa3f5ef97343d6a127790daec

**Description**

During the implementation of the multi-env feature, a `GraviteeContext` based on a thread-local variable has been introduced. Everything works fine a service methods are executed after a call to the rest APIs directly as the Environment and Organization are properly set according to the Path Params. 

BUT when dealing with more complex processes with multiple services calls (like in the login) or asynchronous processes (like with Cockpit commands) the `GraviteeContext` variables aren't usable anymore as they always contain the same env/org. 

To fix that issue this PR starts to remove all calls to `GraviteeContext` from Services, to move them to the Resouce layer. Pretty much all the changes from this PR were done automatically by using IntelliJ refactoring capabilities, meaning it can't have introduced regression (as a drawback, there could have some duplications). Also, it has highlight multiple weird behaviors that will need extra investigation, I've annotated some of them with a `FIXME` and I will add comment on the PR.


**How to tests**

 - All the unit tests should be 🟢 
 - Postman tests should be 🟢 
 - You should not face login issues when the scenario from https://github.com/gravitee-io/issues/issues/6488#issuecomment-995973344: 
    - Linked your APIM installation to cockpit and create multiple env
    - Create an account but don’t log in right now
    - An admin update your user to set you a role to another environment
    - You then try to login 🚀 


🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/remove-current-context-from-services-part-1/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.
